### PR TITLE
Document plugin checks

### PR DESCRIPTION
**Pull request by cushon**

### CODE REVIEW

File: docs/plugins.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    processor_class = "com.google.errorprone.sample.NullAnnotationProcessor",
+    deps = [
+        ":MyErrorPronePluginImplementation",
+        # ...
+    ],
+)
+```
+
+For a complete example, see: 
+http://github.com/google/error-prone/tree/master/examples/plugin/bazel
```
Comment (tbroyer) [inline docs/plugins.md]: nit: use `https`?

File: docs/plugins.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+### Maven
+
+Starting in version 3.5, maven-compiler-plugin allows the processor path to be
+configured with the
+[annotationProcessorPaths](https://maven.apache.org/plugins/maven-compiler-plugin/compile-mojo.html#annotationProcessorPaths)
+parameter.
+
+For a complete example, see: 
+http://github.com/google/error-prone/tree/master/examples/plugin/maven
```
Comment (tbroyer) [inline docs/plugins.md]: Same here

**Comment by tbroyer** [comment]
FWIW, documentation for Gradle: https://github.com/tbroyer/error-prone/commit/70a7d5d77238d52f0c8ebfd80d6a762fb1844f3c
Not sure if I should mention the "manual" way or go straight with plugins. Let's discuss that on the commit itself, or once I send it in a PR (depends on this PR, so waiting for it to be merged first)

File: docs/plugins.md (hunk 1 above)
Comment (cushon) [inline docs/plugins.md]: Done

File: docs/plugins.md (hunk 2 above)
Comment (cushon) [inline docs/plugins.md]: Done

**Comment by tbroyer** [comment]
:shipit:

**Comment by eaftan** [comment]
LGTM

**Pull request merged**